pkg/controller/memberstatus/memberstatus_controller.go

```diff
@@ -291,6 +296,47 @@ func (r *ReconcileMemberStatus) routesHandleStatus(reqLogger logr.Logger, member
 	return nil
 }
 
+// cheIntegrationHandleStatus checks all necessary aspects related integration between the member operator and Che
+// Returns an error if any problems are discovered.
+func (r *ReconcileMemberStatus) cheIntegrationHandleStatus(reqLogger logr.Logger, memberStatus *toolchainv1alpha1.MemberStatus) error {
+	if memberStatus.Status.Che == nil {
+		memberStatus.Status.Che = &toolchainv1alpha1.CheStatus{}
+	}
+
+	// Is che user deletion enabled
+	if !r.config.IsCheUserDeletionEnabled() {
+		// Che user deletion is not enabled, set condition to Ready. No further checks required
+		readyCondition := status.NewComponentReadyCondition(toolchainv1alpha1.ToolchainStatusMemberStatusCheUserDeletionNotEnabledReason)
+		memberStatus.Status.Che.Conditions = []toolchainv1alpha1.Condition{*readyCondition}
```

Comment:
Technically we are doing it wrong. The condition has `LastTransitionTime` which we always update even if there is no change in the condition status. We should update the existing condition (if already exists) instead of replacing it by a new one every time.
But we are doing it wrong for all the other conditions (but the main/overall Ready condition) for all the other components too. It's not critical right now and doesn't break anything though.
So, let's keep it as is and address in a separate PR (let's create a JIRA for that so we can prioritize it properly later).

Reply:
That's a really good point. I guess it has an effect on the status notifications? Opened https://issues.redhat.com/browse/CRT-900

Reply:
Status notifications relies on the overall Ready condition only. So no direct impact on that. For now it's rather about avoiding confusion with that LastTransitionTime timestamps in the component conditions.